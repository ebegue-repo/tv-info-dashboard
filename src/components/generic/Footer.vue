<template>
    <footer class="bg-black text-white">
        <div class="mx-auto max-w-7xl py-12 px-4 sm:px-6 md:flex md:items-center md:justify-between lg:px-8">
            <div class="flex justify-center space-x-6 md:order-2">
                <img src="@/assets/images/tvm_logo.png" />
            </div>
            <div class="mt-8 md:order-1 md:mt-0">
                <p class="text-center text-base text-gray-400">&copy; {{ year }} TV MAZE</p>
            </div>
        </div>
    </footer>
</template>

<script setup lang="ts">
    const year = new Date().getFullYear();
</script>
