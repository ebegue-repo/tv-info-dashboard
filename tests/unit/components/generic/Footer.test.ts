import { mount } from '@vue/test-utils';
import Footer from "@/components/generic/Footer.vue";

describe('FooterVue', () => {
    it('It should render a footer that includes the current year, copyright symbol and the TV MAZE name', () => {

        const wrapper = mount(Footer);
        expect(wrapper.text()).toMatch("\u00A9 " + new Date().getFullYear() + " TV MAZE");
        expect(wrapper.html()).toMatchSnapshot();
    });
});