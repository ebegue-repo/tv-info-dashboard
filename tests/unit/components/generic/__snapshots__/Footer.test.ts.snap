// Vitest Snapshot v1

exports[`FooterVue > It should render a footer that includes the current year, copyright symbol and the TV MAZE name 1`] = `
"<footer class=\\"bg-black text-white\\">
  <div class=\\"mx-auto max-w-7xl py-12 px-4 sm:px-6 md:flex md:items-center md:justify-between lg:px-8\\">
    <div class=\\"flex justify-center space-x-6 md:order-2\\"><img src=\\"/src/assets/images/tvm_logo.png\\"></div>
    <div class=\\"mt-8 md:order-1 md:mt-0\\">
      <p class=\\"text-center text-base text-gray-400\\">Â© 2023 TV MAZE</p>
    </div>
  </div>
</footer>"
`;
